Release build failure with `IggyMessageBus::background_tasks_len`

### Bug description

This accessor here:

- https://github.com/apache/iggy/blob/239d7ee0bd2721c572d38b06669efca21e0c2f6b/core/message_bus/src/lib.rs#L611-L623

...is called by this integration test here: https://github.com/apache/iggy/blob/239d7ee0bd2721c572d38b06669efca21e0c2f6b/core/message_bus/tests/installer_panic_cleanup.rs#L179-L183

When I tried to build a release build of the latest iggy master and I received this compile error:

```
  error[E0599]: no method named `background_tasks_len` found for struct `Rc<IggyMessageBus>` in the current scope
     --> core/message_bus/tests/installer_panic_cleanup.rs:180:13
```

### Deployment

Not applicable

### Versions

master

### Hardware / environment

Nix build

### Sample code

_No response_

### Logs

_No response_

### Iggy server config

_No response_

### Reproduction

_No response_

--- a/core/message_bus/src/lib.rs
+++ b/core/message_bus/src/lib.rs
@@ -623,7 +623,7 @@ impl IggyMessageBus {
     /// callers. A long-running bus is expected to keep this number
     /// bounded under sustained accept traffic; a leak shows up here as
     /// monotonic growth proportional to total accepts.
-    #[cfg(any(test, debug_assertions))]
+    #[doc(hidden)]
     #[must_use]
     pub fn background_tasks_len(&self) -> usize {
         self.background_tasks.borrow().len()

